### PR TITLE
[PROF-10235] Add new metadata fields to elf symbol upload

### PR DESCRIPTION
**Pull request by nsavoire**
Add symbol_source, origin, origin_version and filename to elf symbol uploads.
Remove platform since it's not used on backend side.

### CODE REVIEW

File: symbolication/datadog_uploader.go [hunk 1]
```diff
@@ -147,14 +148,17 @@ func (d *DatadogUploader) HandleExecutable(elfRef *pfelf.Reference, fileID libpf
 }
 
 type executableMetadata struct {
-	Arch       string `json:"arch"`
-	GNUBuildID string `json:"gnu_build_id"`
-	GoBuildID  string `json:"go_build_id"`
-	FileHash   string `json:"file_hash"`
-	Platform   string `json:"platform"`
```
Comment (Gandem) [inline symbolication/datadog_uploader.go:154]: Is the removal of the `Platform` field expected?

**Review by Gandem — APPROVED** [review]
LGTM, thanks 🙇 !

**Comment by Gandem** [comment]
Side note the linter is complaining for line length 😢

**Comment by nsavoire** [comment]
> Side note the linter is complaining for line length 😢

Fixed. This 100 chars limit is annoying

File: symbolication/datadog_uploader.go (hunk 1 above)
Comment (nsavoire) [reply symbolication/datadog_uploader.go:154]: Yes, `platform` is not used on backend side (I also removed it from [datdog-ci](https://github.com/DataDog/datadog-ci/pull/1388)).

**Pull request merged**